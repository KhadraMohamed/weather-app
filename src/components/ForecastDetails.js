import React from "react";
import PropTypes from "prop-types";

function ForecastDetails(props) {
  const { humidity, direction, windspeed, temperature, date } = props;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div className="forecast-details__date"> {formattedDate} </div>
      <div className="forecast-details__humidity"> {humidity} </div>
      <div className="forecast-details__windspeed"> {windspeed} </div>
      <div className="forecast-details__temperature">
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__direction">{direction}</div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  date: PropTypes.number.isRequired,
  humidity: PropTypes.string.isRequired,
  direction: PropTypes.string.isRequired,
  windspeed: PropTypes.string.isRequired,
  temperature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
};
