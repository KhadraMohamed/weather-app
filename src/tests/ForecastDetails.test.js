import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    direction: "Stub direction",
    humidity: "76%",
    windspeed: "15",
    temperature: {
      min: 12,
      max: 22,
    },
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails
        date={validProps.date}
        direction={validProps.direction}
        humidity={validProps.humidity}
        temperature={validProps.temperature}
        windspeed={validProps.windspeed}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("render correct values for props", () => {
    const { getByText } = render(
      <ForecastDetails
        date={validProps.date}
        direction={validProps.direction}
        humidity={validProps.humidity}
        temperature={validProps.temperature}
        windspeed={validProps.windspeed}
      />
    );

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details__date");
    expect(getByText("Stub direction")).toHaveClass(
      "forecast-details__direction"
    );
    expect(getByText("76%")).toHaveClass("forecast-details__humidity");
    expect(getByText("22Â°C")).toHaveClass("forecast-details__temperature");
  });
});
