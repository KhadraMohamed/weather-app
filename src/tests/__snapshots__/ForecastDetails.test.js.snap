// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
       Thu Jan 01 1970 
    </div>
    <div
      class="forecast-details__humidity"
    >
       76% 
    </div>
    <div
      class="forecast-details__windspeed"
    >
       15 
    </div>
    <div
      class="forecast-details__temperature"
    >
      22Â°C
    </div>
    <div
      class="forecast-details__direction"
    >
      Stub direction
    </div>
  </div>
</DocumentFragment>
`;
